### DISCUSSION TITLE
how to check can go back url in capacitor?

### DISCUSSION THREAD

**Discussion by wudobi** [ionic-team/capacitor#4077] [General]
i used capacitor
i wanna know, when i click back button, which url go

so i wrote this code
```javascript
Plugins.App.addListener('backButton', async (data) { // data is always empty, right?
})
```

i think data is url data, but always empty.

if i can check other method (back url), pls teach me xD

**Comment by imhoffd** [comment]
You should be using your JavaScript framework's router for that. Usually, single-page apps have routers that manipulate the URL and store state using the [History API](https://developer.mozilla.org/en-US/docs/Web/API/History). I can't give you more information without knowing more about how your app is built.
